### PR TITLE
Update scripts

### PR DESCRIPTION
**Pull request by maximzubkov**

### CODE REVIEW

File: scripts/download/split_dataset.sh [hunk 1]
```diff
@@ -39,31 +42,45 @@ mkdir "$DIR_TRAIN"
...
-do
-    echo "Splitting class - $DIR_CLASS";
-    mkdir "$DIR_VAL/$DIR_CLASS"
-    mkdir "$DIR_TEST/$DIR_CLASS"
-    num_files=$(find "$DIR_TRAIN/$DIR_CLASS" -type f | wc -l)
+num_files=$(find "$DIR_TRAIN" -mindepth 1 -type d | wc -l)
+
+if [ $DATASET_NAME == "codeforces" ]
+then
+    basenames=$(find "$DIR_TRAIN" -mindepth 1 -type d | \
```
Comment (maximzubkov) [inline scripts/download/split_dataset.sh:51]: 1. `DIR_TRAIN` has the following structure:
```
author_name_1_1312_D_00000001_tests_OK_0001
author_name_2_1313_D_00000002_tests_OK_0002
author_name_2_1312_A_00000003_tests_OK_0003
....
```

File: scripts/download/split_dataset.sh [hunk 2]
```diff
@@ -39,31 +42,45 @@ mkdir "$DIR_TRAIN"
...
+              awk -F '_' '{print $(NF-5)$(NF-4), $0}' | \
```
Comment (maximzubkov) [inline scripts/download/split_dataset.sh:52]: 2. In this line we select tasks, the output is as follows:
```
1312_D author_name_1_1312_D_00000001_tests_OK_0001
1313_D author_name_2_1313_D_00000002_tests_OK_0002
1312_A author_name_2_1312_A_00000003_tests_OK_0003
....
```

File: scripts/download/split_dataset.sh [hunk 3]
```diff
@@ -39,31 +42,45 @@ mkdir "$DIR_TRAIN"
...
+              sort -k1 | \
```
Comment (maximzubkov) [inline scripts/download/split_dataset.sh:53]: 3. Then we sort these lines by the tasks and get:
```
1312_A author_name_2_1312_A_00000003_tests_OK_0003
1312_D author_name_1_1312_D_00000001_tests_OK_0001
1313_D author_name_2_1313_D_00000002_tests_OK_0002
....
```

File: scripts/download/split_dataset.sh [hunk 4]
```diff
@@ -39,31 +42,45 @@ mkdir "$DIR_TRAIN"
...
+              cut -d ' ' -f2-)
```
Comment (maximzubkov) [inline scripts/download/split_dataset.sh:54]: 4. Finally, we cut the task name, leaving only `dirs`:
```
author_name_2_1312_A_00000003_tests_OK_0003
author_name_1_1312_D_00000001_tests_OK_0001
author_name_2_1313_D_00000002_tests_OK_0002
....
```

File: scripts/download/split_dataset.sh [hunk 5]
```diff
@@ -39,31 +42,45 @@ mkdir "$DIR_TRAIN"
...
+    bounds=$(python $COMPUTE_BOUNDS_SCRIPT "$DIR_TRAIN" "$TRAIN_SPLIT_PART" "$TEST_SPLIT_PART")
```
Comment (maximzubkov) [inline scripts/download/split_dataset.sh:55]: 5. Bounds are defined in using python script and returned as a string

**Review by maximzubkov — COMMENTED** [review]
As we discussed at the meeting, it is necessary to divide the data on `train/test/val` in such a way that each competition is completely contained in only one holdout. To achieve this goal I (1) sort all the dirs with data by the competition id and (2) define `train_bound` and `test_bound` which are the indexes defining the boundary of `train` and `test` set 

To make it clear, I explain each step precisely

**Review by SpirinEgor — COMMENTED** [review]
Code is fine. I suggest adding your comments directly in the script. So in the future, we wouldn't forget what is happening in splitting dataset logic.

**Review by SpirinEgor — APPROVED** [review]

**Pull request merged**